Solutions/Web Session Essentials/Hunting Queries/KaliLinuxUserAgentDetected.yaml

```diff
@@ -0,0 +1,41 @@
+id: 05f5cccd-ee93-4293-b7ad-05820aaa54a4
+name: Detect Kali Linux UserAgent (ASIM Web Session)
+description: |
+  'This rule helps to detect usage of Kali Linux in your environment. Attackers might utilize Kali Linux's tools and features for unauthorized penetration testing, reconnaissance, or exploitation attempts.'
+tags:
+  - Schema: WebSession
+    SchemaVersion: 0.2.6
+requiredDataConnectors: []
+tactics:
+  - Execution
+relevantTechniques:
+  - T1059
+  - T1203
+query: |
+  let lookback = 1d;
+  _Im_WebSession (starttime=ago(lookback))
+  | project HttpUserAgent, SrcIpAddr, SrcUsername, SrcHostname, Url, TimeGenerated
+  | where HttpUserAgent has_all ("Linux", "Kali")
```

Comment:
We should use httpuseragent_has_any filter for Linux and Kali and then filter here if both are present. This would improve the performance.